agent/dockerclient/clientfactory/dockerclientfactory.go

```diff
@@ -180,17 +180,13 @@ func getDockerClientForVersion(
 	minAPIVersion string,
 	apiVersion string) (dockeriface.Client, error) {
 	if minAPIVersion != "" && apiVersion != "" {
-		// Adding patch number zero to Docker versions to reuse the existing semver
-		// comparator
-		// TODO: remove this logic later when non-semver comparator is implemented
-		versionWithPatch := version + zeroPatch
-		lessThanMinCheck := "<" + minAPIVersion + zeroPatch
-		moreThanMaxCheck := ">" + apiVersion + zeroPatch
-		minVersionCheck, err := utils.Version(versionWithPatch).Matches(lessThanMinCheck)
+		lessThanMinCheck := "<" + minAPIVersion
```

Comment:
i think this variable should be deleted now: https://github.com/aws/amazon-ecs-agent/blob/b166bcbc74242ff7c7efd04466ea06eb368069f4/agent/dockerclient/clientfactory/dockerclientfactory.go#L35